--- a/drivers/gpu/drm/nouveau/dispnv50/disp.c	
+++ a/drivers/gpu/drm/nouveau/dispnv50/disp.c	
@@ -785,10 +785,10 @@ nv50_hdmi_enable(struct drm_encoder *encoder, struct nouveau_crtc *nv_crtc,
 	max_ac_packet -= rekey;
 	max_ac_packet -= 18; /* constant from tegra */
 	max_ac_packet /= 32;
+	u8 scdc = 0;
 
 	if (nv_encoder->i2c && hdmi->scdc.scrambling.supported) {
 		const bool high_tmds_clock_ratio = mode->clock > 340000;
-		u8 scdc;
 
 		ret = drm_scdc_readb(nv_encoder->i2c, SCDC_TMDS_CONFIG, &scdc);
 		if (ret < 0) {
@@ -809,7 +809,7 @@ nv50_hdmi_enable(struct drm_encoder *encoder, struct nouveau_crtc *nv_crtc,
 	}
 
 	ret = nvif_outp_hdmi(&nv_encoder->outp, nv_crtc->index, true, max_ac_packet, rekey,
-			     mode->clock, hdmi->scdc.supported, hdmi->scdc.scrambling.supported,
+			     mode->clock, scdc, hdmi->scdc.scrambling.supported,
 			     hdmi->scdc.scrambling.low_rates);
 	if (ret)
 		return;
